File: db/fsharp/SalesManagement.Domain/Models/OrderDetail.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 受注データ明細
/// </summary>
[<CLIMutable>]
type OrderDetail = {
    OrderNo: string                // 受注番号（PK、FK）
    OrderLineNo: int               // 受注行番号（PK）
    ProductCode: string            // 商品コード（FK）
    ProductName: string            // 商品名（履歴保持）
    Quantity: int                  // 受注数量
    AllocatedQuantity: int         // 引当数量
    ShippingInstructionQuantity: int  // 出荷指示数量
    ShippedQuantity: int           // 出荷済数量
    UnitPrice: int                 // 売単価（履歴保持）
    Amount: int                    // 金額
    ConsumptionTax: int            // 消費税
    IsCompleted: int               // 完了フラグ (0: 未完了, 1: 完了)
    CreatedAt: DateTime            // 作成日時
    CreatedBy: string              // 作成者名
    UpdatedAt: DateTime            // 更新日時
    UpdatedBy: string              // 更新者名
}

File: db/csharp/SalesManagement.Domain/Models/OrderDetail.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 受注データ明細
    /// </summary>
    public class OrderDetail
    {
        public string OrderNo { get; set; } = string.Empty;
        public int OrderRowNo { get; set; }
        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public int UnitPrice { get; set; }
        public int Quantity { get; set; }
        public int ConsumptionTaxRate { get; set; }
        public int ReserveQuantity { get; set; }
        public int DeliveryOrderQuantity { get; set; }
        public int DeliveredQuantity { get; set; }
        public int CompleteFlag { get; set; }
        public int Discount { get; set; }
        public DateTime? DeliveryDate { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}
